### PR TITLE
Karate refactoring part 2

### PR DESCRIPTION
**Pull request by simone-kalbermatter**
This PR refactors the following features:
- create.feature
- createRollCall.feature
- openRollCall.feature
- closeRollCall.feature

The main changes are:
- The existing class `MultiMsgWebSocketClient` uses a randomly generated key pair instead of the hardcoded one
- Created a `mockClient` class that extends `MultiMsgWebSocketClient` and adds the functionality to dynamically create Lao and Roll Call objects that can be used in tests
- Replaced the current `mockFronted.feature` with a new `mockClient.feature` that allows creation of many objects of the `mockClient` class in other features

### CODE REVIEW

**Review by matteosz — DISMISSED** [review]
LGTM!

File: tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java [hunk 1]
```diff
@@ -148,7 +153,21 @@ public boolean receiveNoMoreResponses(){
...
+
+  /**
+   * Set the client to use a wrong signature when sending messages
+   */
+  public void useWrongSignature() {
+    String wrongSignature = RandomUtils.generateSignature();
+    logger.info("setting wrong signature: " + wrongSignature);
+    jsonConverter.setSignature(wrongSignature);
   }
+
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java:172]: Slight formatting mistake 😉

File: tests/karate/src/test/java/be/utils/GoServer.java [hunk 2]
```diff
@@ -20,13 +20,13 @@ public String[] getCmd() {
...
         "/c",
-        "\"pop.exe server --pk J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= serve\""
+        "\"pop.exe server serve\""
       };
     } else {
       return new String[]{
         "bash",
         "-c",
-        "./pop server --pk J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM= serve"
+        "./pop server serve"
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/utils/GoServer.java:29]: I have a question :
How is the organizer key given to the server without this argument ?

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
I am truly amazed by the work you've done !

This is so much better and on top of that, the code is clear and documented.

I was only able to find a formatting mistake 😅

I also have a question regarding a modification you did when starting the Go Server.

File: tests/karate/src/test/java/be/utils/GoServer.java (hunk 2 above)
Comment (GabrielFleischer) [reply tests/karate/src/test/java/be/utils/GoServer.java:29]: Oh ok, that's something I didn't follow. Thanks for the clarification.

**Review by GabrielFleischer — APPROVED** [review]
LGTM

**Pull request merged**